flux-common/src/main/java/software/amazon/aws/clients/swf/flux/poller/TaskNaming.java

```diff
@@ -107,9 +107,54 @@ public static String createActivityId(String stepName, long retryAttempt, String
     }
 
     /**
-     * Given a step name, constructs a partition metadata marker name.
+     * Given a step name and information about how many metadata markers there are,
+     * constructs a partition metadata marker name.
      */
-    public static String partitionMetadataMarkerName(String stepName) {
-        return String.format("%s.%s", PARTITION_METADATA_MARKER_NAME_PREFIX, stepName);
+    public static boolean isPartitionMetadataMarker(String markerName) {
+        return markerName != null && markerName.startsWith(PARTITION_METADATA_MARKER_NAME_PREFIX);
+    }
+
+    /**
+     * Given a step name and information about how many metadata markers there are,
+     * constructs a partition metadata marker name.
+     */
+    public static String partitionMetadataMarkerName(String stepName, int subsetId, int markerCount) {
```

Comment:
If we are parsing things out of this string as Longs, should these inputs also be longs?

Reply:
For consistency, yeah, probably. I'll fix it. In practice they're going to be single-digit numbers though.